kafka-connect-s3/src/main/java/io/confluent/connect/s3/hooks/BlockingKafkaPostCommitHook.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Add here
+ */
+
+package io.confluent.connect.s3.hooks;
+
+import io.confluent.connect.s3.S3SinkConnectorConfig;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.KafkaException;
+import org.apache.kafka.common.errors.AuthorizationException;
+import org.apache.kafka.common.errors.OutOfOrderSequenceException;
+import org.apache.kafka.common.errors.ProducerFencedException;
+import org.apache.kafka.common.errors.UnsupportedVersionException;
+import org.apache.kafka.common.header.internals.RecordHeader;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.kafka.connect.errors.RetriableException;
+
+import java.util.Properties;
+import java.util.Set;
+
+public class BlockingKafkaPostCommitHook implements PostCommitHook {
+
+  private static final Logger log = LoggerFactory.getLogger(BlockingKafkaPostCommitHook.class);
+  private String kafkaTopic;
+  private KafkaProducer<String, String> kafkaProducer;
+
+  @Override
+  public void init(S3SinkConnectorConfig config) {
+    if (kafkaProducer != null) {
```

Comment:
why do we need to check this condition ?

Reply:
not needed, removed